x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/servicenow_params.tsx

```diff
@@ -88,8 +88,18 @@ const ServiceNowParamsFields: React.FunctionComponent<
     if (!actionParams.subAction) {
       editAction('subAction', 'pushToService', index);
     }
+    if (!actionParams.subActionParams) {
+      editAction(
+        'subActionParams',
+        {
+          incident: {},
+          comments: [],
+        },
+        index
+      );
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [actionParams]);
```

Comment:
need to run `editAction` with `subActionParams` on mount in order to run error validation, but setting both with deps of `[]` creates a race condition and only one gets set. by checking for undefined values whenever the actionParams change, we can overcome the race condition